apps/crn-server/src/data-providers/contentful/news.data-provider.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  ListNewsDataObject,
+  NewsDataObject,
+  FetchNewsFilter,
+  FetchPaginationOptions,
+} from '@asap-hub/model';
+
+import {
+  getClient,
+  ContentfulRestNews,
+  News,
+  parseRichText,
+} from '@asap-hub/contentful';
+
+export interface NewsDataProvider {
+  fetchById(id: string): Promise<NewsDataObject | null>;
+  fetch: (options?: FetchNewsProviderOptions) => Promise<ListNewsDataObject>;
+}
+
+export class NewsContentfulDataProvider {
+  constructor(private contentfulClient: ReturnType<typeof getClient>) {}
+
+  async fetch(options?: FetchNewsProviderOptions) {
+    try {
+      const { total, items } = await this.contentfulClient.getEntries<News>({
+        content_type: 'news',
+        'fields.title[match]': options?.filter?.title,
+        'fields.frequency[in]': options?.filter?.frequency?.join(','),
+        limit: options?.take,
+        skip: options?.skip,
+        order: '-sys.createdAt',
+      });
+
+      return {
+        total,
+        items: items.map(parsetNews),
+      };
+    } catch (error) {
+      if (JSON.parse((error as Error).message)?.status === 404) {
```

Comment:
this is strange. This is a fetch method so why would we look for a 404 at all here? I

Reply:
There is a test for this in the squidex data provider https://github.com/yldio/asap-hub/blob/master/apps/crn-server/test/data-providers/news.data-provider.test.ts#L67, I thought we would want the same behaviour